Mathlib/Algebra/Homology/DerivedCategory/Basic.lean

```diff
@@ -0,0 +1,188 @@
+* [Mark Hovey, *Model category structures on chain complexes of sheaves*][hovey-2001]
+
+-/
+
+universe w v u
+
+open CategoryTheory Limits
+
+variable (C : Type u) [Category.{v} C] [Abelian C]
+
+namespace HomotopyCategory
+
+/-- The triangulated subcategory of `HomotopyCategory C (ComplexShape.up ℤ)` consisting
+of acyclic complexes. -/
+def subcategoryAcyclic : Triangulated.Subcategory (HomotopyCategory C (ComplexShape.up ℤ)) :=
+  (homologyFunctor C (ComplexShape.up ℤ) 0).homologicalKernel
+
+instance : ClosedUnderIsomorphisms (subcategoryAcyclic C).P := by
+  dsimp [subcategoryAcyclic]
+  infer_instance
+
+variable {C}
+
+lemma mem_subcategoryAcyclic_iff (X : HomotopyCategory C (ComplexShape.up ℤ)) :
+    (subcategoryAcyclic C).P X ↔ ∀ (n : ℤ), IsZero ((homologyFunctor _ _ n).obj X) :=
+  Functor.mem_homologicalKernel_iff _ X
+
+lemma quotient_obj_mem_subcategoryAcyclic_iff_exactAt (K : CochainComplex C ℤ) :
+    (subcategoryAcyclic C).P ((quotient _ _).obj K) ↔ ∀ (n : ℤ), K.ExactAt n := by
+  rw [mem_subcategoryAcyclic_iff]
+  refine forall_congr' (fun n => ?_)
+  simp only [HomologicalComplex.exactAt_iff_isZero_homology]
+  exact ((homologyFunctorFactors C (ComplexShape.up ℤ) n).app K).isZero_iff
+
+variable (C)
+
+lemma quasiIso_eq_subcategoryAcyclic_W :
+    quasiIso C (ComplexShape.up ℤ) = (subcategoryAcyclic C).W := by
+  ext K L f
+  exact ((homologyFunctor C (ComplexShape.up ℤ) 0).mem_homologicalKernel_W_iff f).symm
+
+end HomotopyCategory
+
+/-- The assumption that a localized category for
+`(HomologicalComplex.quasiIso C (ComplexShape.up ℤ))` has been chosen, and that the morphisms
+in this chosen category are in `Type w`. -/
+abbrev HasDerivedCategory := MorphismProperty.HasLocalization.{w}
+  (HomologicalComplex.quasiIso C (ComplexShape.up ℤ))
+
+/-- The derived category obtained using the constructed localized category of cochain complexes
+with respect to quasi-isomorphisms. This should be used only while proving statements
+which do not involve the derived category. -/
+def HasDerivedCategory.standard : HasDerivedCategory.{max u v} C :=
+  MorphismProperty.HasLocalization.standard _
+
+variable [HasDerivedCategory.{w} C]
+
+/-- The derived category of an abelian category. -/
+def DerivedCategory := HomologicalComplexUpToQuasiIso C (ComplexShape.up ℤ)
+
+namespace DerivedCategory
+
+instance : Category (DerivedCategory C) := by
+  dsimp [DerivedCategory]
+  infer_instance
+
+variable {C}
+
+/-- The localization functor `CochainComplex C ℤ ⥤ DerivedCategory C`. -/
+def Q : CochainComplex C ℤ ⥤ DerivedCategory C := HomologicalComplexUpToQuasiIso.Q
+
+instance : (Q (C := C)).IsLocalization
+    (HomologicalComplex.quasiIso C (ComplexShape.up ℤ)) := by
+  dsimp only [Q, DerivedCategory]
+  infer_instance
+
+/-- The localization functor `HomotopyCategory C (ComplexShape.up ℤ) ⥤ DerivedCategory C`. -/
+def Qh : HomotopyCategory C (ComplexShape.up ℤ) ⥤ DerivedCategory C :=
+  HomologicalComplexUpToQuasiIso.Qh
```

Comment:
Do we want to make `C` explicit in `Q` and `Qh`? Seems that there are lots of `Q (C := C)` below.

Reply:
Once the suitable instances are in place (for which we need this `Q (C := C)`, we shall mostly write `Q.obj K` with `K : CochainComplex C _`. Then, the `C` can be easily infered.